### PR TITLE
fix: scan plugins-scanning directory at startup as well

### LINKED ISSUE #1681: Activate extension from 'plugin-scanning' folder on podman-desktop start up
### Bug description

The extensions copied to 'plugin-scanning' folder are activated only when podman desktop is already running.  After restarting podman desktop extension from 'plugin-scanning' folder are not activated.

### Operating system

macOS Ventura 13.2 (Apple M1)

### Version

0.12.0

### Steps to reproduce

1. Start podman desktop
2. Copy packed extension to `.local/share/podman-desktop/plugins-scanning`
3. See extension is getting activated and present in `Settings/Extensions`
4. Restart podman desktop
5. See extension from `.local/share/podman-desktop/plugins-scanning` is not present in `Settings/Extensions`

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
scan the directory at startup and not only when there are changes on the folder

Also add unit tests

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop/issues/1681

### How to test this PR?

1. copy a .cdix (extension file) after startup in `plugins-scanning` folder
2. check extension is loaded
3. restart Podman Desktop and check extension is loaded again

Change-Id: I22dd380c79f9b6c3e66467f36a328896ab665e44

### CODE REVIEW

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Seems the expect import is missing in the unit test

**Comment by benoitf** [comment]
@jeffmaury fixed

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Tests are failing on Windows due to path conversions

**Comment by benoitf** [comment]
@jeffmaury ok, I tried something different, let me know

**Comment by jeffmaury** [comment]
> @jeffmaury ok, I tried something different, let me know

Stlll have 1 error:

image

**Comment by benoitf** [comment]
@jeffmaury ok, I tried something..I'm like in the dark

File: packages/main/src/plugin/extension-loader.spec.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  readdirMock.mockResolvedValue([ent1, ent2, ent3]);
+
+  // mock loadPackagedFile
+  const loadPackagedFileMock = vi.spyOn(extensionLoader, 'loadPackagedFile');
+  loadPackagedFileMock.mockResolvedValue();
+
+  await extensionLoader.setupScanningDirectory();
+
+  // expect to load only one file (other are invalid files/folder)
+  expect(loadPackagedFileMock).toBeCalledWith(path.join(rootedFakeDirectory, 'foo.cdix'));
```
Comment (jeffmaury) [inline packages/main/src/plugin/extension-loader.spec.ts]:
```suggestion
  expect(loadPackagedFileMock).toBeCalledWith(path.join(fakeDirectory, 'foo.cdix'));
```

Fix Windows test failure

**Comment by benoitf** [comment]
suggestion applied

File: packages/main/src/plugin/extension-loader.spec.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  );
+});
+
+beforeEach(() => {
+  vi.clearAllMocks();
+});
+
+test('Should watch for files and load them at startup', async () => {
+  const fakeDirectory = '/fake/path/scanning';
+  const rootedFakeDirectory = path.resolve(fakeDirectory);
```
Comment (benoitf) [inline packages/main/src/plugin/extension-loader.spec.ts]:
```suggestion

```

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**